atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepableCellsReadWriteTest.java

```diff
@@ -0,0 +1,89 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static com.palantir.atlasdb.sweep.queue.ShardAndStrategy.conservative;
+import static com.palantir.atlasdb.sweep.queue.ShardAndStrategy.thorough;
+
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.atlasdb.keyvalue.api.Cell;
+
+public class SweepableCellsReadWriteTest extends SweepQueueReadWriteTest {
+    private SweepableCellsReader reader;
+
+    int shard;
+    int shard2;
+
+    @Before
+    public void setup() {
+        super.setup();
+        writer = new SweepableCellsWriter(kvs, partitioner);
+        reader = new SweepableCellsReader(kvs);
+
+        shard = writeTs(writer, TS, true);
+        shard2 = writeTs(writer, TS2, false);
+    }
+
+    @Test
+    public void canReadSingleEntryInSingleShard() {
+        assertThat(reader.getLatestWrites(TS_REF, conservative(shard)))
+                .containsExactly(WriteInfo.of(TABLE_REF, Cell.create(ROW, COL), TS));
+        assertThat(reader.getLatestWrites(TS2_REF, thorough(shard2)))
+                .containsExactly(WriteInfo.of(TABLE_REF2, Cell.create(ROW, COL), TS2));
+    }
+
+    @Test
+    public void getOnlyMostRecentTimestampForCellAndTableRef() {
+        writeTs(writer, TS - 1, true);
+        writeTs(writer, TS + 2, true);
+        writeTs(writer, TS - 2, true);
+        writeTs(writer, TS + 1, true);
+        assertThat(reader.getLatestWrites(TS_REF, conservative(shard)))
+                .containsExactly(WriteInfo.of(TABLE_REF, Cell.create(ROW, COL), TS + 2));
+    }
+
+    @Test
+    public void canReadMultipleEntriesInSingleShardDifferentTransactions() {
+        int fixedShard = writeTsToCell(writer, TS, getCellWithFixedHash(1), true);
+        assertThat(writeTsToCell(writer, TS + 1, getCellWithFixedHash(2), true)).isEqualTo(fixedShard);
+        assertThat(reader.getLatestWrites(TS_REF, conservative(fixedShard))).containsExactlyInAnyOrder(
+                WriteInfo.of(TABLE_REF, getCellWithFixedHash(1), TS),
+                WriteInfo.of(TABLE_REF, getCellWithFixedHash(2), TS + 1));
+    }
+
+    @Test
+    public void canReadMultipleEntriesInSingleShardSameTransactionNotDedicated() {
+        List<WriteInfo> writes = writeToUniqueCellsInSameShard(writer, TS, 10, true);
+        ShardAndStrategy fixedShardAndStrategy = conservative(WriteInfoPartitioner.getShard(writes.get(0)));
+        assertThat(writes.size()).isEqualTo(10);
+        assertThat(reader.getLatestWrites(TS_REF, fixedShardAndStrategy)).hasSameElementsAs(writes);
+    }
+
+    @Test
+    public void canReadMultipleEntriesInSingleShardSameTransactionDedicated() {
+        List<WriteInfo> writes = writeToUniqueCellsInSameShard(writer, TS, 257, true);
```

Comment:
Doesn't the premise of this test depend on `257` being above the limit of dedicated writes? Is there a way to assert that the thing we wrote was a dedicated write? If not, is it nice to use "threshold + 1" rather than 257 for the number of cells?